src/Microsoft.Extensions.FileProviders.Physical/PollingFileChangeToken.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Physical
+{
+    public class PollingFileChangeToken : IChangeToken
+    {
+        private FileInfo _fileInfo;
+        private DateTime _previousWriteTimeUtc;
+        private DateTime _lastCheckedTimeUtc;
+        private bool _hasChanged;
+
+        public PollingFileChangeToken(FileInfo fileInfo)
+        {
+            _fileInfo = fileInfo;
+            _previousWriteTimeUtc = GetLastWriteTimeUtc();
```

Comment:
no, no. I'm talking about something different. I'm talking about the file being deleted between the moment when you invoke the constructor and the moment when this line of code runs. Basically if the file is deleted and never recreated you'll never know that it was deleted. My suggestion was just to make this as "_hasChanged" if the file doesn't exist when you create the object - it means that it existed just before